### PR TITLE
Cleanup IFont related interop

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Cleanup IFont related interop code
- Add IFont/IPicture tests

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/1939)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AxHostTests.cs [hunk 1]
```diff
@@ -948,6 +950,300 @@ public void AxHost_GetOcx_NotCreated_ReturnsNull()
...
+            Assert.Equal(PixelFormat.Format32bppRgb, result.PixelFormat);
+            Assert.Equal(Color.FromArgb(unchecked((int)0xFF010203)), original.GetPixel(1, 2));
+
+            result = Assert.IsType<Bitmap>(SubAxHost.GetPictureFromIPicture(iPictureDisp));
+            Assert.Equal(original.Size, result.Size);
+            Assert.Equal(PixelFormat.Format32bppRgb, result.PixelFormat);
+            Assert.Equal(Color.FromArgb(unchecked((int)0xFF010203)), original.GetPixel(1, 2));
+        }
+
+#if false
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AxHostTests.cs]: What's the intention with this one?

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AxHostTests.cs [hunk 2]
```diff
@@ -948,6 +950,300 @@ public void AxHost_GetOcx_NotCreated_ReturnsNull()
...
+
+            IPictureDisp iPictureDisp = (IPictureDisp)disp;
+            Assert.NotNull(iPictureDisp);
+            Assert.NotEqual(0u, iPictureDisp.Handle);
+            Assert.Equal(0u, iPictureDisp.hPal);
+            Assert.Equal(1, iPictureDisp.Type);
+            Assert.Equal(265, iPictureDisp.Width);
+            Assert.Equal(291, iPictureDisp.Height);
+            //Assert.Equal(0u, iPictureDisp.CurDC);
+            //Assert.Equal(0u, iPictureDisp.Attributes);
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AxHostTests.cs]: Remove?

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AxHostTests.cs [hunk 3]
```diff
@@ -999,6 +1295,147 @@ public new ImeMode ImeModeBase
...
+
+            public new static object GetIPictureFromPicture(Image image) => AxHost.GetIPictureFromPicture(image);
+
+            public new static Image GetPictureFromIPicture(object picture) => AxHost.GetPictureFromIPicture(picture);
         }
+
+        [ComImport]
+        [Guid("7BF80980-BF32-101A-8BBB-00AA00300CAB")]
+        [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+        private interface IPicture
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AxHostTests.cs]: Is there a reason we need to duplicate `Interop.Ole32.IPicture`?

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AxHostTests.cs (hunk 1 above)
Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AxHostTests.cs]: Accidental

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AxHostTests.cs (hunk 2 above)
Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AxHostTests.cs]: Removed

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AxHostTests.cs (hunk 3 above)
Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AxHostTests.cs]: See comment: https://github.com/dotnet/winforms/blob/master/src/Common/src/Interop/Ole32/Interop.IPicture.cs#L33

**Comment by hughbe** [comment]
@JeremyKuhne Is this a bug? Can IDispatch interfaces not use BOOL?

image

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AxHostTests.cs (hunk 3 above)
Comment (RussKie) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AxHostTests.cs]: Thank you.

I suggest we add a comment there to effect of:
```cs
/*
 *  We currently don't use the rest of the methods here. Keeping them undefined to make the RCW smaller.
 *  If this changes we need to remove a duplicate definition in AxHostTests
 *
```

...and add a comment here
```cs
// A duplicate as Interop.Ole32.IPicture is only partially implemented to make RCW smaller
```

This way anyone modifying the code is fully aware of the duplication.


The same applies to other dup interfaces.

File: src/Common/src/UnsafeNativeMethods.cs [hunk 4]
```diff
@@ -2596,94 +2589,6 @@ public interface IOleDocumentView
...
-            void QueryTextMetrics(out IntPtr ptm);
-
-            void AddRefHfont(
-                    IntPtr hFont);
-
-            void ReleaseHfont(
-                    IntPtr hFont);
-
-            void SetHdc(
-                    IntPtr hdc);
```
Comment (weltkante) [inline src/Common/src/UnsafeNativeMethods.cs]: @hughbe not sure if my initial comment on the commit notified you so I'll also drop a note here on the PR, `QueryTextMetrics` and `SetHdc` got lost in the port, I created #2007 to keep track of it. No rush, just want to make sure you are aware of it.

**Pull request merged**